papaya-web/src/main/java/pl/edu/agh/papaya/rest/sprints/controller/SprintsController.java

```diff
@@ -0,0 +1,117 @@
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import pl.edu.agh.papaya.api.model.SprintDto;
+import pl.edu.agh.papaya.api.model.SprintStateDto;
+import pl.edu.agh.papaya.mappers.LocalDateTimePeriodMapper;
+import pl.edu.agh.papaya.mappers.SprintMapper;
+import pl.edu.agh.papaya.mappers.SprintStateMapper;
+import pl.edu.agh.papaya.model.LocalDateTimePeriod;
+import pl.edu.agh.papaya.model.Project;
+import pl.edu.agh.papaya.model.Sprint;
+import pl.edu.agh.papaya.model.SprintState;
+import pl.edu.agh.papaya.service.sprint.SprintService;
+import pl.edu.agh.papaya.util.AssertionUtil;
+import pl.edu.agh.papaya.util.ResourceNotFoundException;
+
+@SuppressWarnings({"PMD.BeanMembersShouldSerialize"})
+@RestController
+@RequiredArgsConstructor
+public class SprintsController {
+
+    private static final List<SprintStateDto> ALL_SPRINT_STATE_DTOS = Lists.newArrayList(SprintStateDto.values());
+
+    private final SprintService sprintService;
+
+    private final SprintStateMapper sprintStateMapper;
+
+    private final SprintMapper sprintMapper;
+
+    public ResponseEntity<List<SprintDto>> getSprints(Project project, @Valid List<SprintStateDto> sprintStateDtos) {
+        final LocalDateTime currentTime = LocalDateTime.now();
+
+        List<SprintState> sprintStates = sprintStateMapper.mapFromApi(
+                Optional.ofNullable(sprintStateDtos).orElse(ALL_SPRINT_STATE_DTOS));
+
+        List<SprintDto> sprints = sprintService.getByStatesInProject(sprintStates, project.getId(), currentTime)
+                .stream()
+                .sorted(Comparator.comparing(sprint -> sprint.getEnrollmentPeriod().getStart()))
+                .map(sprint -> sprintMapper.mapToApi(sprint, currentTime))
+                .collect(Collectors.toList());
+
+        return ResponseEntity.ok(sprints);
+    }
+
+    public ResponseEntity<SprintDto> addSprint(Project project, @Valid SprintDto sprintDto) {
+        var periodMapper = new LocalDateTimePeriodMapper();
+        LocalDateTimePeriod enrollmentPeriod = periodMapper.mapFromApi(sprintDto.getEnrollmentPeriod());
+        LocalDateTimePeriod durationPeriod = periodMapper.mapFromApi(sprintDto.getDurationPeriod());
+
+        Optional<Sprint> lastSprintOpt = sprintService.getLastInProject(project.getId());
+        lastSprintOpt.ifPresent(lastSprint -> {
+            if (!enrollmentPeriod.isAfter(lastSprint.getDurationPeriod())) {
+                throw new IllegalStateException("Cannot create sprint starting before the last one ended");
+            }
+        });
+
+        @SuppressWarnings({"MultipleStringLiterals"})
+        Long timePlanned = AssertionUtil.requirePositive("timePlanned", sprintDto.getTimePlanned());
```

Comment:
Ja tutaj postuluję `nonNegative` jednak

Reply:
Aktualnie w widoczku nie ma tego pola w sumie i teraz kliknięcie tutaj `+ Stwórz` rzuca wyjątki :stuck_out_tongue: 

Pole na te godziny ma być dodane, ale i tak pozwoliłabym tutaj na 0 - to są tylko godziny planowane i można je potem zmieniać

![obraz](https://user-images.githubusercontent.com/34098778/57412789-4bd84600-71f2-11e9-8659-9ec5d1d55100.png)